daemon/daemon.go

```diff
@@ -361,6 +363,20 @@ configRetry:
 		simulateDataRace()
 	}
 
+	// Set source-destination-check on AWS EC2 instance.
```

Comment:
IIUC, we'll want to do this in a component hooked into the calculation graph rather than part of daemon.go.

@fasaxc does that match your understanding? Feels like we at least want it to be driven off of config updates post validation filter. And, potentially, handle these changes without triggering a felix restart. Feels like it doesn't need to go into the data plane though, since it's agnostic to which data plane is being used. WDYT?

Reply:
This is currently invoked after the configuration/validation and the outcome of src-dst-check operation doesn't cause  Felix to restart.

Reply:
Will this code work unmodified on Windows?  If not, it'll break when merged over to enterprise.  I'd be inclined to move it to the dataplane driver since it is part of setting up the dataplane (and that will prevent it from breaking on Windows).  If it's needed by multiple dataplane types then they can call through to a utility function that does this work.   (But it's probably not needed for all dataplane types that we may ever support.)

Felix normally restarts when its config changes; I'm not sure I see a reason to do the extra work for this config parameter to handle it without restart.  The main issue is that needs a lot more testing.

Should we back off to an occasional retry instead of trying to filter retriable/non-retriable errors?  I'm thinking a non-retriable error could be something like a permissions error; but the user could fix a permissions error at any time.  OTOH, if you have a large cluster with thousands of nodes hitting the API, that could cause throttling.